### PR TITLE
Http2 frames 5743 v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/5743
https://redmine.openinfosecfoundation.org/issues/7213

Describe changes:
- http2: support frames
- frames: fix use of stream frame (only when enabled)

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2011

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22121

File: src/app-layer-parser.c [hunk 1]
```diff
@@ -1231,7 +1231,8 @@ static void HandleStreamFrames(Flow *f, StreamSlice stream_slice, const uint8_t
             }
         }
     } else if (flags & STREAM_EOF) {
-        Frame *frame = AppLayerFrameGetById(f, direction, FRAME_STREAM_ID);
+        // if we are not using FRAME_STREAM_TYPE, FRAME_STREAM_ID points to another frame
```
Comment (victorjulien) [inline src/app-layer-parser.c:1234]: this talks about FRAME_STREAM_ID, but that is removed in the PR, so it's not a valid reference

Comment (catenacyber) [reply src/app-layer-parser.c:1234]: right, the commit message seems enough

**Comment by catenacyber** [comment]
Continues in https://github.com/OISF/suricata/pull/11707

**Pull request closed without merging**